src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1423,9 +1425,48 @@ internal bool IsSubmissionSyntaxTree(SyntaxTree tree)
         /// <summary>
         /// Global imports (including those from previous submissions, if there are any).
         /// </summary>
-        internal Imports GlobalImports => _globalImports.Value;
+        internal ImmutableArray<NamespaceOrTypeAndUsingDirective> GlobalImports => _globalImports.Value;
 
-        private Imports BindGlobalImports() => Imports.FromGlobalUsings(this);
+        private ImmutableArray<NamespaceOrTypeAndUsingDirective> BindGlobalImports()
+        {
+            var usingsFromoptions = UsingsFromOptions;
+            var previousSubmission = PreviousSubmission;
```

Comment:
>var previousSubmission = PreviousSubmission; [](start = 12, length = 44)

From here, this is what Imports.FromGlobalUsings used to do at the end, after it processed usings from options.